src/menu.scss

```diff
@@ -5,165 +5,303 @@
+        color: $fd-menu-primary-icon-color;
+      }
 
       @include fd-hover() {
-        @include fd-var-color("background-color", fd-color-state("selected-hover"), --fd-color-background-selected-hover);
+        background-color: $fd-menu-item-background-color--selected-hover;
       }
     }
 
-    @include fd-active-pressed-selected() {
-      @include fd-var-color("color", fd-color-state("selected", "action"), --fd-color-action-selected);
-
+    &:focus {
+      box-shadow: none;
       outline: none;
+      outline-offset: -0.0625rem;
+      outline-style: var(--sapContent_FocusStyle);
+      outline-width: var(--sapContent_FocusWidth);
+      outline-color: var(--sapContent_FocusColor);
     }
 
-    &:focus {
-      outline-style: dotted;
-      outline-width: 1px;
-
-      @include fd-var-color("outline-color", fd-color-state("hover", "action"), --fd-color-action-focus);
+    @include fd-disabled() {
+      opacity: var(--sapContent_DisabledOpacity);
+      cursor: not-allowed;
+      text-decoration: none;
     }
 
-    @include fd-disabled() {
-      @include fd-var-color("outline-color", fd-color-state("disabled", "action"), --fd-color-action-disabled);
-      @include fd-var-color("color", $fd-forms-color--disabled, --fd-color-text-4);
+    @include fd-item-responsive-padding();
 
-      cursor: not-allowed;
+    .#{$block}__option-name {
+      @include fd-reset();
+
+      flex: 3 3 10%;
+      color: $fd-menu-item-color;
+
+      @include fd-weight("normal");
+
+      font-size: var(--sapFontSize);
+      font-family: var(--sapFontFamily);
       text-decoration: none;
+
+      @include fd-ellipsis();
+    }
+  }
+
+  @mixin icon-before-size-responsive() {
+    @include fd-screen(s) {
+      width: 2.75rem; // to account for item right padding of 16px or 1rem so 2.75rem-1rem = 1.75rem
+      height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+      margin: 0 0 0 -1rem;
+
+      @include fd-rtl() {
+        margin: 0 -1rem 0 0; // since before and after icons switch places, their margins also switch
+      }
+    }
+
+    @include fd-screen(m) {
+      width: 2.25rem; // to account for item right padding of 16px or 1rem so 2.25rem-1rem = 1.25rem
+      height: 1.25rem; // to account for item top padding of 12px or 0.75rem so 2.75rem-0.75rem-0.75rem = 1.25rem
+      margin: 0 0 0 -0.75rem;
+
+      @include fd-rtl() {
+        margin: 0 -0.1rem 0 0; // since before and after icons switch places, their margins also switch
+      }
+    }
+
+    @include fd-screen(l) {
+      width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
+      height: 1.25rem; // to account for item top padding of 6px or 0.375rem so 2rem-0.375rem-0.375rem = 1.25rem
+      margin: 0 0 0 -0.5rem;
+
+      @include fd-rtl() {
+        margin: 0 -0.75rem 0 0; // since before and after icons switch places, their margins also switch
+      }
```

Comment:
Let's not use `fd-screen`, we let users to add size modes by themselves on `fd-menu` level. so ex. `fd-menu--compact`.

Reply:
I was handling cozy, compact modes for tablet, desktop and mobile through this. If we provide option to user to add size modes, then we won't have control if they decide to use compact mode on phone thereby not aligning to Fiori3, right?

Reply:
Actually, in all of cases, when we have more than 1 `size mode` such as cozy/compact, we let users decide on application level, which one they want to use it.

Reply:
Actually, in all of cases, when we have more than 1 `size mode` such as cozy/compact, we let users decide on application level, which one they want to use it.

Reply:
added `fd-menu--compact-desktop` and `fd-menu--mobile` classes. Tablet cozy is default.